File: jules-scratch/verification/verify_tutorial.py
import re
from playwright.sync_api import sync_playwright, Page, expect

def run(playwright):
    browser = playwright.chromium.launch(headless=True)
    context = browser.new_context()
    page = context.new_page()

    try:
        # 1. Navigate and Login
        page.goto("http://localhost:8080")

        expect(page.locator("#login-form")).to_be_visible(timeout=10000)
        page.locator("#login-email").fill("f.santoro@barackmercosul.com")
        page.locator("#login-password").fill("$oof@k24")
        page.get_by_role("button", name="Iniciar Sesión").click()

        # 2. Wait for app to load
        expect(page.locator("#view-title")).to_have_text("Dashboard", timeout=20000)

        # 3. Start the tutorial
        page.get_by_role("button", name="ECO/ECR").click()
        start_tutorial_button = page.get_by_role("link", name="Iniciar Tutorial")
        expect(start_tutorial_button).to_be_visible()
        start_tutorial_button.click()

        # 4. Verify the tutorial steps
        tutorial_tooltip = page.locator("#tutorial-tooltip")
        next_button = tutorial_tooltip.get_by_role("button", name="Siguiente")

        # Step 1: Welcome
        expect(tutorial_tooltip.get_by_role("heading")).to_have_text("Bienvenido al Tutorial de Gestión PRO", timeout=5000)
        next_button.click()

        # Step 2: Menu. The postAction for this step opens the dropdown.
        expect(tutorial_tooltip.get_by_role("heading")).to_have_text("Módulo ECO/ECR", timeout=5000)
        next_button.click()

        # Step 3: Go to ECR view. The postAction for this step switches the view.
        # The element for this step is now visible because the dropdown is open.
        expect(tutorial_tooltip.get_by_role("heading")).to_have_text("Gestión de ECR", timeout=5000)
        next_button.click()

        # Step 4: ECR Screen. The view should now have changed.
        expect(page.locator("#view-title")).to_have_text("Gestión de ECR", timeout=10000)
        expect(tutorial_tooltip.get_by_role("heading")).to_have_text("Pantalla de Gestión de ECR", timeout=5000)
        page.screenshot(path="jules-scratch/verification/verification_final.png")

        print("Playwright script executed successfully. Tutorial is fixed.")

    except Exception as e:
        print(f"An error occurred: {e}")
        page.screenshot(path="jules-scratch/verification/error.png")
    finally:
        browser.close()

with sync_playwright() as playwright:
    run(playwright)

File: verify_tutorial.js
import { chromium } from 'playwright';
import assert from 'assert';

(async () => {
    const browser = await chromium.launch();
    const page = await browser.newPage();

    try {
        console.log('Navigating to the application...');
        await page.goto('http://localhost:8080');

        console.log('Waiting for login form...');
        await page.waitForSelector('#login-form', { timeout: 15000 });

        console.log('Logging in...');
        await page.fill('#login-email', 'f.santoro@barackmercosul.com');
        await page.fill('#login-password', '$oof@k24');
        await page.click('button[type="submit"]');

        console.log('Waiting for app view to be visible...');
        await page.waitForSelector('#app-view:not(.hidden)', { timeout: 15000 });
        console.log('Login successful.');

        console.log('Opening the ECO/ECR dropdown menu...');
        await page.click('[data-tutorial-id="eco-ecr-menu"] button');

        console.log('Waiting for the "Iniciar Tutorial" button...');
        const startTutorialButton = await page.waitForSelector('#start-tutorial-btn', { timeout: 5000 });

        console.log('Starting the tutorial...');
        await startTutorialButton.click();

        console.log('Waiting for the tutorial overlay to appear...');
        await page.waitForSelector('#tutorial-overlay', { state: 'visible', timeout: 5000 });
        console.log('Tutorial started successfully.');

        // --- Verification Step 1: Check for improved UI (progress indicator) ---
        console.log('Verifying Step 1: Checking for progress indicator...');
        await page.waitForSelector('#tutorial-tooltip-progress', { timeout: 2000 });
        const progressText = await page.textContent('#tutorial-tooltip-progress');
        assert.strictEqual(progressText, 'Paso 1 de 14', `Expected 'Paso 1 de 14' but got '${progressText}'`);
        console.log('Progress indicator found and correct.');

        // Take a screenshot of the first step to show the new UI
        const screenshotPath = 'public/tutorial_step1_verified.png';
        await page.screenshot({ path: screenshotPath });
        console.log(`Screenshot saved to ${screenshotPath}`);

        // --- Verification Step 2: Navigate through a few steps ---
        console.log('Navigating to step 2...');
        await page.click('#tutorial-next-btn');
        await page.waitForSelector('.nav-dropdown.open', { timeout: 2000 }); // Wait for menu to open

        console.log('Navigating to step 3...');
        await page.click('#tutorial-next-btn');
        await page.waitForSelector('a[data-view="ecr"].tutorial-click-effect', { timeout: 2000 });

        console.log('Navigating to step 4...');
        await page.click('#tutorial-next-btn');
        // Wait for the tutorial to advance to the correct step
        await page.waitForFunction(() => document.getElementById('tutorial-tooltip-progress')?.textContent === 'Paso 4 de 14', { timeout: 5000 });
        const finalProgressText = await page.textContent('#tutorial-tooltip-progress');
        assert.strictEqual(finalProgressText, 'Paso 4 de 14', `Expected 'Paso 4 de 14' but got '${finalProgressText}'`);
        console.log('Successfully navigated a few steps.');

        console.log('Frontend verification successful!');

    } catch (error) {
        console.error('Verification failed:', error);
        // Take a screenshot on error for debugging
        await page.screenshot({ path: 'public/error_screenshot.png' });
        throw error; // Re-throw the error to fail the process
    } finally {
        await browser.close();
    }
})();
